lib/test_suite/bird_count.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ElixirAnalyzer.TestSuite.BirdCount do
+  @dialyzer generated: true
+  @moduledoc """
+  This is an exercise analyzer extension module for the concept exercise Bird Count
+  """
+
+  alias ElixirAnalyzer.Constants
+
+  use ElixirAnalyzer.ExerciseTest
+
+  not_allowed_functions =
+    [Enum, List, Stream]
+    |> Enum.flat_map(fn module ->
+      Enum.map(module.module_info(:exports), fn {fun, _arity} -> {module, fun} end)
+    end)
+
+  Enum.map(not_allowed_functions, fn {module, function} ->
+    assert_no_call "does not call #{module}.#{function}" do
+      type :essential
+      called_fn module: module, name: function
+      comment Constants.bird_count_use_recursion()
+    end
+  end)
```

Comment:
@neenjaw I'm unable to get this to work. I'm getting this error:

```
$ mix test test/elixir_analyzer/test_suite/bird_count_test.exs
Compiling 1 file (.ex)

== Compilation error in file lib/test_suite/bird_count.ex ==
** (ArgumentError) calling function signature requires :module to be nil or a module atom, got: {:module, [line: 20], nil}
    lib/elixir_analyzer/exercise_test/assert_call.ex:121: ElixirAnalyzer.ExerciseTest.AssertCall.validate_module/1
    lib/elixir_analyzer/exercise_test/assert_call.ex:105: ElixirAnalyzer.ExerciseTest.AssertCall.validate_signature/1
    lib/elixir_analyzer/exercise_test/assert_call.ex:67: ElixirAnalyzer.ExerciseTest.AssertCall.do_walk_assert_call_block/2
    (elixir 1.11.3) lib/macro.ex:482: anonymous fn/4 in Macro.do_traverse_args/4
    (elixir 1.11.3) lib/enum.ex:1533: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (elixir 1.11.3) lib/enum.ex:1533: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (elixir 1.11.3) lib/macro.ex:448: Macro.do_traverse/4
    lib/elixir_analyzer/exercise_test/assert_call.ex:57: ElixirAnalyzer.ExerciseTest.AssertCall.walk_assert_call_block/2
```

I'm sure it's something obvious and I'm probably attempting to use macros incorrectly. I tried various combinations of `quote/unquote` and I just don't get it (I'm having a bad day probably...) Could you advise? How do I achieve what I need? (Dynamically pass all Enum, List, and Stream functions to `assert_no_call`)

Reply:
I arrived at this monster: [`77ced7a` (#50)](https://github.com/exercism/elixir-analyzer/pull/50/commits/77ced7a2dd6fcfb423df73beec20b9019a47c4d4)